dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/model/ResourceLink.java

```diff
@@ -0,0 +1,165 @@
+        this.fileAsset = fileAsset;
+        this.editableAsText = editableAsText;
+        this.downloadRestricted = downloadRestricted;
+    }
+
+    public String getResourceLinkAsString() {
+        return resourceLinkAsString;
+    }
+
+    public String getResourceLinkUriAsString() {
+        return resourceLinkUriAsString;
+    }
+
+    public String getMimeType() {
+        return mimeType;
+    }
+
+    public FileAsset getFileAsset() {
+        return fileAsset;
+    }
+
+    public boolean isEditableAsText() {
+        return editableAsText;
+    }
+
+    public boolean isDownloadRestricted() {
+        return downloadRestricted;
+    }
+
+    @Override
+    public String toString() {
+        return resourceLinkAsString;
+    }
+
+    public static class ResourceLinkBuilder {
+
+        public final ResourceLink build(final HttpServletRequest request, final User user, final Contentlet contentlet) throws DotDataException, DotSecurityException {
+
+            if(!(contentlet.getContentType() instanceof FileAssetContentType)){
+                throw new DotStateException("Can only build Resource Links out of content with type `File Asset`.");
+            }
+
+            final StringBuilder resourceLink = new StringBuilder();
+            final StringBuilder resourceLinkUri = new StringBuilder();
+
+            final Identifier identifier = getIdentifier(contentlet);
+            final Host host = getHost((String)request.getAttribute(HOST_REQUEST_ATTRIBUTE) , user);
+            if (identifier != null && InodeUtils.isSet(identifier.getInode())){
+                if(request.isSecure()){
+                    resourceLink.append(HTTPS_PREFIX);
+                }else{
+                    resourceLink.append(HTTP_PREFIX);
+                }
+                resourceLink.append(host.getHostname());
+                if(request.getServerPort() != 80 && request.getServerPort() != 443){
+                    resourceLink.append(StringPool.COLON).append(request.getServerPort());
+                }
+                resourceLinkUri.append(identifier.getParentPath()).append(contentlet.getStringProperty(FileAssetAPI.FILE_NAME_FIELD));
+                resourceLink.append(UtilMethods.encodeURIComponent(resourceLinkUri.toString()));
+                resourceLinkUri.append(LANG_ID_PARAM).append(contentlet.getLanguageId());
+                resourceLink.append(LANG_ID_PARAM).append(contentlet.getLanguageId());
+            }
+
+            final FileAsset fileAsset = getFileAsset(contentlet);
+            final String mimeType = fileAsset.getMimeType();
+            final String fileAssetName = fileAsset.getFileName();
+
+            return new ResourceLink(resourceLink.toString(), resourceLinkUri.toString(), mimeType, fileAsset, isEdiatbleAsText(mimeType, fileAssetName), isDownloadRestricted(fileAssetName));
+        }
+
+        private static boolean isEdiatbleAsText(final String mimeType, final String fileAssetName ){
```

Comment:
Small typo in the method's name

Reply:
That's on purpose dah!   just kidding I'll fix it.

Reply:
Fixed.